errors/src/commonMain/kotlin/dev/icerock/moko/errors/handler/ExceptionHandler.kt

```diff
@@ -4,19 +4,6 @@
 
 package dev.icerock.moko.errors.handler
 
-import dev.icerock.moko.errors.ErrorEventListener
-import dev.icerock.moko.errors.ErrorPresenter
-import dev.icerock.moko.mvvm.dispatcher.EventsDispatcher
-
-class ExceptionHandler(
-    errorPresenter: ErrorPresenter,
-    private val errorEventsDispatcher: EventsDispatcher<ErrorEventListener>
-) : ExceptionHandlerBinder by ExceptionHandlerBinderImpl(
-    errorPresenter,
-    errorEventsDispatcher
-) {
-
-    fun <T> handle(block: suspend () -> T): ExceptionHandlerContext<T> {
-        return ExceptionHandlerContext(errorEventsDispatcher, block)
-    }
+interface ExceptionHandler : ExceptionHandlerBinder {
+    fun <R> handle(block: suspend () -> R): ExceptionHandlerContext<R>
 }
```

Comment:
i think companion object way will be more clear in public api:
```
interface ExceptionHandler : ExceptionHandlerBinder {
    fun <R> handle(block: suspend () -> R): ExceptionHandlerContext<R>

    companion object {
        operator fun <T : Any> invoke(
            errorPresenter: ErrorPresenter<T>,
            errorEventsDispatcher: EventsDispatcher<ErrorEventListener<T>>,
            onCatch: ((Throwable) -> Unit)? = null
        ): ExceptionHandler {
            return ExceptionHandlerImpl(
                errorPresenter = errorPresenter,
                errorEventsDispatcher = errorEventsDispatcher,
                onCatch = onCatch
            )
        }
    }
}

internal class ExceptionHandlerImpl<T : Any>(
    private val errorPresenter: ErrorPresenter<T>,
    private val errorEventsDispatcher: EventsDispatcher<ErrorEventListener<T>>,
    private val onCatch: ((Throwable) -> Unit)? = null
) : ExceptionHandler, ExceptionHandlerBinder by ExceptionHandlerBinderImpl<T>(
    errorPresenter,
    errorEventsDispatcher
) {
    override fun <R> handle(block: suspend () -> R): ExceptionHandlerContext<R> {
        return ExceptionHandlerContext(errorPresenter, errorEventsDispatcher, block)
    }
}
```

nobody will not search what `ExceptionHandler` we should create...just will use creation from companion:
```
fun test() {
    val h: ExceptionHandler = ExceptionHandler(
        errorPresenter = AlertErrorPresenter(), 
        errorEventsDispatcher = EventsDispatcher() 
    )
}
```

Reply:
Reworked.